### DISCUSSION TITLE
install real editble text file

### DISCUSSION THREAD

**Discussion by germgerm** [NuGet/Home#12320] [General]
I know this has been asked many times but I have yet to find any useful answer
This is something that I need to do in many packages, so the work arounds I have found are just too involved for more than a one off.
I need to have a text file installed in the consuming project that is a real, editable file preferably in the bin folder alongside the .dll
The closest I have gotten is an unusable linked file in the app root using target=content/contentFiles
Unusable because : 
1 - the file is unmodifiable - or rather any modifications made are made globally for all projects using the package and lost when the package is updated.  Also the file cannot be deleted or excluded.
2 - the file is marked as Item Type=Compile, which it is not and prevents the project from building 

Using target=lib similar to the .dll, the file is packaged alongside the .dll in the .nupkg file but it is not installed into the consuming project anywhere

**Comment by zivkan** [comment]
The short version is that it's not possible.

Due to the design of PackageReference, especially given that lock files did not exist when PackageReference was first implemented, there's no way to detect "package being restored for the first time" vs "project already had this PackageReference, but this is the first time restoring after a git clean". This is also due to the fact that PackageReference was designed to be hand editable in any text editor, not requiring special tools like the dotnet cli or Visual Studio to install packages.

While NuGet can be used to distribute [(dotnet cli) templates](https://learn.microsoft.com/en-us/dotnet/core/tools/custom-templates), installing templates is done with `dotnet new`. I believe Visual Studio detects all `dotnet new` templates and makes them available in the GUI as well, but I'm not 100% sure about that.

**Comment by germgerm** [comment]
Seems to me that nuget needs a redesign

This was my solution - based on the fact that all of our packages are dependant upon one core package
Create an assembly level custom attribute
Have packages include the text file as an embedded resource and mark the assembly with our custom attribute
Have our config handler in the core package check all of our dlls in the bin folder for this custom attribute - if found, extract the embedded file from the assembly - actually we just get the embedded resource stream and never actually extract the file.
Currently we are only concerned with one json file ( appsettings.json) , but the scheme can handle additional items if required
